mozjs: new spack package for SpiderMonkey (aka Mozilla JavaScript)

KineticTheory opened the pull request:
<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

SpiderMonkey is Mozilla’s JavaScript and WebAssembly Engine, used in Firefox, Servo and various other projects. It is written in C++, Rust and JavaScript. You can embed it into C++ and Rust projects, and it can be run as a stand-alone shell.

* [x] Requires #1913

KineticTheory commented:
@spackbot fix style

KineticTheory commented:
@spackbot re-run pipeline

rbberger commented on repos/spack_repo/builtin/packages/mozjs/package.py:
```diff
@@ -0,0 +1,44 @@
+
+    maintainers("KineticTheory")
+
+    license("Mozilla Public License, v. 2.0")
```
should be
```suggestion
    license("MPL-2.0")
```
see https://spdx.dev/

rbberger commented on repos/spack_repo/builtin/packages/mozjs/package.py:
```diff
@@ -0,0 +1,44 @@
+    build_directory = "spack-build"
+
+    def configure_args(self):
+        args = ["--disable-jemalloc", "--with-system-zlib", "--with-intl-api", "--enable-optimize"]
```
@KineticTheory have you tried with `depends_on("zlib-api")` and without `--with-system-zlib`?

KineticTheory replied:
That works for me.  I've updated the recipe.

KineticTheory committed:
mozjs: new spack package for SpiderMonkey (aka Mozilla JavaScript)

SpiderMonkey is Mozilla’s JavaScript and WebAssembly Engine, used in Firefox, Servo and various other projects. It is written in C++, Rust and JavaScript. You can embed it into C++ and Rust projects, and it can be run as a stand-alone shell.

KineticTheory committed:
This doesn't currently work on MacOS.
